packages/client/src/utils/contract.ts

```diff
@@ -139,8 +139,16 @@ export const createDecoratedContract = <T extends Contract>(
     const result: any = {
         eventEmitter
     }
+
+    function getAllPropertyNames(obj: object): string[] {
+        const proto = Object.getPrototypeOf(obj)
+        const inherited = (proto) ? getAllPropertyNames(proto) : []
+        return [...new Set(Object.getOwnPropertyNames(obj).concat(inherited))]
+    }
+
     // copy own properties and inherited properties (e.g. contract.removeAllListeners)
-    for (const key in contract) {
+    // eslint-disable-next-line no-prototype-builtins
+    for (const key of getAllPropertyNames(contract)) {
```

Comment:
Is this needed?

Reply:
Is needed, otherwise in browser [end-to-end tests](https://github.com/streamr-dev/network/actions/runs/7712499768/job/21020246975):

```
1) event listener: picks up add and remove events
     StorageNodeRegistry
     TypeError: opts.sourceEmitter.on is not a function
    at /tmp/_karma_webpack_610790/commons.js:66166:24
    at EventEmitter.<anonymous> (/tmp/_karma_webpack_610790/commons.js:100933:18)
    at EventEmitter.emit (/tmp/_karma_webpack_610790/commons.js:133517:33)
    at StreamrClientEventEmitter.on (/tmp/_karma_webpack_610790/commons.js:99667:21)
    at StreamrClient.on (/tmp/_karma_webpack_610790/commons.js:53192:25)
    at UserContext.<anonymous> (/tmp/_karma_webpack_610790/commons.js:68473:20)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/tmp/_karma_webpack_610790/commons.js:68399:103)
    at _next (/tmp/_karma_webpack_610790/commons.js:68400:194)

2) listener
     StreamRegistry createStream
     TypeError: opts.sourceEmitter.on is not a function
    at /tmp/_karma_webpack_610790/commons.js:66166:24
    at EventEmitter.<anonymous> (/tmp/_karma_webpack_610790/commons.js:100933:18)
    at EventEmitter.emit (/tmp/_karma_webpack_610790/commons.js:133517:33)
    at StreamrClientEventEmitter.on (/tmp/_karma_webpack_610790/commons.js:99667:21)
    at StreamrClient.on (/tmp/_karma_webpack_610790/commons.js:53192:25)
    at UserContext.<anonymous> (/tmp/_karma_webpack_610790/commons.js:68726:14)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/tmp/_karma_webpack_610790/commons.js:68652:103)
    at _next (/tmp/_karma_webpack_610790/commons.js:68653:194)
    at /tmp/_karma_webpack_610790/commons.js:68653:364
```